homeassistant/components/fritz/common.py

```diff
@@ -331,11 +331,19 @@ def scan_devices(self, now: datetime | None = None) -> None:
         _LOGGER.debug("Checking host info for FRITZ!Box device %s", self.host)
         self._update_available, self._latest_firmware = self._update_device_info()
 
-        try:
-            topology = self.fritz_hosts.get_mesh_topology()
-        except FritzActionError:
-            self.mesh_role = MeshRoles.SLAVE
-            return
+        if (
+            "Hosts1" not in self.connection.services
```

Comment:
Device details should be handled by the library.

Reply:
This is not a low level call, I don't think the library should be responsible for it.
We have the list of available services for each device, and based on that we decide what to do on HA side.

Simone

Reply:
The point of the library is that it should handle device variations and abstract that.

Reply:
So you would like to have a method like `mesh_supported()` in the library to replace this if ?

Reply:
Yes, that sounds good. 👍

Reply:
Let's see if the author of the library agree; anyway it will slow down a lot the fix :-(

Reply:
I am going to merge this PR for the hot fix.

We can update using the new function from the lib once it's been released.